File: plugin_files/bin/scripting/0_sdk.js
globalThis.sdk = new Proxy({}, {
    get: (obj, key, receiver) => {
        if (IsTypeClass(key)) return GenerateTypeFactory(key);
        else if (IsSDKClass(key)) return (ptr) => {
            let f = GenerateSDKFactory(key, ptr)
            return f;
        }
        else return undefined;
    },
    set: (target, key, val, recv) => {
        throw SyntaxError("You cannot set values to SDK.");
    }
})

File: plugin_files/bin/scripting/0_sdk.lua
_G = setmetatable(_G, {
    __index = function(t, k)
        if IsSDKClass(k) then
            return function(ptr)
                return GenerateSDKFactory(k, ptr)
            end
        elseif IsTypeClass(k) then
            return GenerateTypeFactory(k)
        end
    end
})

sdk = setmetatable({}, {
    __index = function(t, k)
        if IsSDKClass(k) then
            return function(ptr)
                return GenerateSDKFactory(k, ptr)
            end
        elseif IsTypeClass(k) then
            return GenerateTypeFactory(k)
        end
    end
})